.github/workflows/ci.yml

```
@@ -0,0 +1,237 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - main
+      - '*.x'
+    tags:
+      - '*'
+  pull_request:
+    branches:
+      - main
+  schedule:
+    # Weekly Monday 9AM build
+    # * is a special character in YAML so you have to quote this string
+    - cron: '0 9 * * 1'
+
+env:
+  CRDS_SERVER_URL: "https://roman-crds-test.stsci.edu"
+  CRDS_PATH: "/tmp/crds_cache"
+  CRDS_CLIENT_RETRY_COUNT: 3
+  CRDS_CLIENT_RETRY_DELAY_SECONDS: 20
+
+jobs:
+  style:
+    name: Code style checks
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3.0.2
```

```suggestion
      - uses: actions/checkout@v3
```

I would only pin to the major version (no minor, or bugfix version pins). @zacharyburnett what do you think?

---

yes indeed, it would be better to be `v3`

---

Done